PWGJE/TableProducer/jettaggerhfExtension.cxx

```diff
@@ -0,0 +1,197 @@
+  AxisSpec DcaXYAxis = {binDcaXY, "DCA_{XY} [#mum]"};
+  AxisSpec DcaZAxis = {binDcaZ, "DCA_{Z} [#mum]"};
+  AxisSpec DcaXYZAxis = {binDcaXYZ, "DCA_{XYZ} [#mum]"};
+  AxisSpec DcaXYSigmaAxis = {binDcaXY, "#sigma_{DCA_{XY}} [#mum]"};
+  AxisSpec DcaXYZSigmaAxis = {binDcaXY, "#sigma_{DCA_{XYZ}} [#mum]"};
+
+  int runNumber = -1;
+
+  HistogramRegistry registry{"registry", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  void init(o2::framework::InitContext& initContext)
+  {
+    ccdb->setURL(ccdburl);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    lut = o2::base::MatLayerCylSet::rectifyPtrFromFile(ccdb->get<o2::base::MatLayerCylSet>(lutPath));
+
+    if (doFillHistogram) {
+      registry.add("hDCAx", "DCA to PV; DCA_{x} (#mum)", {HistType::kTH1F, {DcaXAxis}});
+      registry.add("hDCAy", "DCA to PV; DCA_{y} (#mum)", {HistType::kTH1F, {DcaYAxis}});
+      registry.add("hDCAz", "DCA to PV; DCA_{z} (#mum)", {HistType::kTH1F, {DcaZAxis}});
+      registry.add("hDCAxy", "DCA to PV; DCA_{xy} (#mum)", {HistType::kTH1F, {DcaXYAxis}});
+      registry.add("hDCAxyz", "DCA to PV; DCA_{xyz} (#mum)", {HistType::kTH1F, {DcaXYZAxis}});
+      registry.add("hDCAxyVsDCAxyFromExt", "DCA to PV; DCA_{xy} (#mum); DCA_{xy}^{ext} (#mum)", {HistType::kTH2F, {{DcaXYAxis}, {DcaXYAxis}}});
+      registry.add("hDCAxy_uncertainty", "uncertainty of DCA to PV; #sigma_{DCA_{xy}} (#mum)", {HistType::kTH1F, {DcaXYSigmaAxis}});
+      registry.add("hDCAxyz_uncertainty", "uncertainty of DCA to PV; #sigma_{DCA_{xyz}} (#mum)", {HistType::kTH1F, {DcaXYZSigmaAxis}});
+    }
+  }
+
+  template <typename T>
+  void initCCDB(T const& bc)
+  {
+    if (runNumber == bc.runNumber()) {
+      return;
+    }
+    grpmag = ccdb->getForTimeStamp<o2::parameters::GRPMagField>(grpmagPath, bc.timestamp());
+    LOG(info) << "Setting magnetic field to current " << grpmag->getL3Current() << " A for run " << bc.runNumber() << " from its GRPMagField CCDB object";
+    o2::base::Propagator::initFieldFromGRP(grpmag);
+    o2::base::Propagator::Instance()->setMatLUT(lut);
+    mVtx = ccdb->getForTimeStamp<o2::dataformats::MeanVertexObject>(mVtxPath, bc.timestamp());
+    runNumber = bc.runNumber();
+  }
+
+  void calculateDcaXYZ(float& dcaXYZ, float& sigmaDcaXYZ2, float dcaXY, float dcaZ, float cYY, float cZY, float cZZ, float sigmaDcaXY2, float sigmaDcaZ2)
+  {
+    dcaXYZ = std::sqrt(dcaXY * dcaXY + dcaZ * dcaZ);
+    float dFdxy = 2 * dcaXY / dcaXYZ;
+    float dFdz = 2 * dcaZ / dcaXYZ;
+    sigmaDcaXYZ2 = std::abs(cYY * dFdxy * dFdxy + cZZ * dFdz * dFdz + 2 * cZY * dFdxy * dFdz);
+  }
+
+  void processDummy(aod::Collision const& collision)
+  {
+  }
+  PROCESS_SWITCH(JetTaggerHFTrackExtension, processDummy, "Dummy process", true);
+
+  void processTracks(soa::Join<aod::Tracks, aod::TracksCov, aod::TrackSelection, aod::TracksDCA, aod::TracksDCACov, aod::McTrackLabels>::iterator const& track, aod::BCsWithTimestamps&, aod::Collisions&)
```

Comment:
what exactly is this task doing? It seems to be recalculating the DCA of tracks but it is doing so with respect to the original collision? dont these coloumns already exist?